app/views/devise/registrations/new.html.erb

```diff
@@ -23,6 +23,11 @@
         <%= f.password_field :password_confirmation, autocomplete: 'new-password', class: 'form-control' %>
       </div>
 
+      <div class="form-group">
+        <%= f.label :location %>
+        <%= f.text_field :location, class: 'form-control', autofocus: true %>
+      </div>
```

Comment:
Try extracting the repeating parts of code between `new` and `edit` to a partial. You can use `if` to show / hide soem fields if they are slightly different between them.